x/bucket/internal/ipfs_generate_bucket.go

```diff
@@ -0,0 +1,28 @@
+package internal
+
+import (
+	"context"
+	"fmt"
+
+	shell "github.com/ipfs/go-ipfs-api"
+	"github.com/sonr-io/sonr/pkg/did"
+	bt "github.com/sonr-io/sonr/x/bucket/types"
+)
+
+func GenerateBucket(sh *shell.Shell, b *bt.Bucket, address string) (*did.Service, error) {
+	path := b.GetPath(address)
+	service := b.GetDidService(address)
+	err := sh.FilesMkdir(context.Background(), path, shell.FilesWrite.Create(true))
```

Comment:
i think this will remake the top level `address` directory as well if a bucket has already been created with this address.